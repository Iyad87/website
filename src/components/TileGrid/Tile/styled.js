import styled from 'styled-components';
import breakpoints from 'styles/breakpoints';

import {
  highlight,
  highlightDark,
  highlightDarker,
  textOnDark,
} from 'styles/colours';

import { spacing } from 'styles/spacing';

const colours = {
  normal: highlight,
  dark: highlightDark,
  darker: highlightDarker,
};

const textColours = {
  normal: textOnDark,
  dark: textOnDark,
  darker: textOnDark,
};

export const Container = styled.li`
  background-color: ${props => colours[props.colour]};
  color: ${props => textColours[props.colour]};
  display: flex;
  position: relative;
  min-height: 400px;
  width: 100%;

  ${breakpoints.desktop`
    width: 50%;
  `};
`;

export const Content = styled.div`
  bottom: 0;
  left: 0;
  padding: ${spacing(4)}
  position: absolute;
  text-align: bottom;
`;

export const ContentTitle = styled.h5`
  font-weight: 500;
  font-size: 24px;
  margin: 0;
`;

export const ContentBody = styled.p`
  font-weight: 300;
  font-size: 24px;
  margin: 0;
`;
