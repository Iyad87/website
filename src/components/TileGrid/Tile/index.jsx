import React from 'react';
import PropTypes from 'prop-types';
import { Container, Content, ContentTitle, ContentBody } from './styled';

const Tile = ({ colour }) => (
  <Container colour={colour}>
    <Content>
      <ContentTitle>hello</ContentTitle>
      <ContentBody>hello, world! i represent some tiled content</ContentBody>
    </Content>
  </Container>
);

Tile.propTypes = {
  colour: PropTypes.oneOf(['normal', 'dark', 'darker']),
};

Tile.defaultProps = {
  colour: 'normal',
};

export default Tile;
