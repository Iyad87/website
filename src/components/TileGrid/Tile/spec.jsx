import React from 'react';
import { shallow } from 'enzyme';
import Tile from './index';

test('components/TileGrid/Tile snapshot', () => {
  const wrapper = shallow(
    <Tile title="Hello, world">This is some example content.</Tile>,
  );

  expect(wrapper).toMatchSnapshot();
});
