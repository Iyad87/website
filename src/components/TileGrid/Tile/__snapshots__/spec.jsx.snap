// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TileGrid/Tile snapshot 1`] = `
<styled.li
  colour="normal"
>
  <styled.div>
    <styled.h5>
      Hello, world
    </styled.h5>
    <styled.p>
      This is some example content.
    </styled.p>
  </styled.div>
</styled.li>
`;
